src/jdbc/cubrid/jdbc/driver/CUBRIDConnection.java

```diff
@@ -988,8 +988,14 @@ public String getClientInfo(String arg0) throws SQLException {
 	}
 
 	/* JDK 1.6 */
-	public boolean isValid(int arg0) throws SQLException {
-		throw new SQLException(new java.lang.UnsupportedOperationException());
+	public boolean isValid(int timeout) throws SQLException {
+		if (u_con == null || is_closed) return false;
+
+		try {
+			return u_con.isValid(timeout);
+		} catch (SQLException e) {
+			return false;
+		}
```

Comment:
According to JDBC Spec, SQLException should be thrown if the value supplied for timeout is less then 0.